### DISCUSSION TITLE
Is this a windows issue or something else?

### DISCUSSION THREAD

**Discussion by mentalizer** [antiwork/shortest#198] [Q&A] [answered]
I can run Shortest on a linux machine, but when trying to run it on windows I get this:

`$ npx shortest

📄 C:\Users\K\Documents\Shortest\app\__tests__\login.test.ts
X [ERROR] Failed to create output directory: mkdir C:\Users\K\AppData\Local\Temp\shortest-cache\C:: The filename, directory name, or volume label syntax is incorrect.

X [ERROR] Failed to create output directory: mkdir C:\Users\K\AppData\Local\Temp\shortest-cache\C:: The filename, directory name, or volume label syntax is incorrect.

Test Execution Error: Build failed with 2 errors:
error: Failed to create output directory: mkdir C:\Users\K\AppData\Local\Temp\shortest-cache\C:: The filename, directory name, or volume label syntax is incorrect.
error: Failed to create output directory: mkdir C:\Users\K\AppData\Local\Temp\shortest-cache\C:: The filename, directory name, or volume label syntax is incorrect.
error: Failed to create output directory: mkdir C:\Users\K\AppData\Local\Temp\shortest-cache\C:: The filename, directory name, or volume label syntax is incorrect.
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
 name, or volume label syntax is incorrect.
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 Tests    0 passed (0)
 Duration   0.81s
 Start at   3:00:19 AM

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯`

Tried patching, adding the cache directory as well etc, asked Cursor for help, but I can't run this on my windows.

Would love to hear if others had the same experience..

I have all my tests running on my pc, and wanted to add shortest to my current implementation.

**Comment by slavingia** [comment]
May be an issue with Windows, yeah. We merged something in recently if you look at recent commits, but it isn't tied to a version bump yet.

**Comment by adks3489** [comment]
I met the same problem with Windows.

I try to fix it by edit the `bin.js`, There will be two problem.

```js
import { basename } from "path";
  async compileFile(filePath) {
    const fileName = basename(filePath).replace(".ts", ".mjs"); // orig: const fileName = filePath.split("/").pop().replace(".ts", ".mjs");
```

```js
  async executeTestFile(file) {
...
      await import(`file://${compiledPath}`); // orig: await import(`${compiledPath}`);
```

**Comment by m2rads** [comment]
Would be a good idea to open an issue and we can start working on it.

**Comment by slavingia** [comment]
For stuff like this best to fix yourself and submit a PR; as most devs don't have Windows.

**Comment by rmarescu** [comment, marked as answer]
Should be fixed in [v0.4.2](https://github.com/anti-work/shortest/releases/tag/v0.4.2)!
